core/shared/src/main/scala/zio/FiberRefs.scala

```diff
@@ -26,50 +27,76 @@ import scala.annotation.tailrec
  * example between an asynchronous producer and consumer.
  */
 final class FiberRefs private (
-  private[zio] val fiberRefLocals: Map[FiberRef[_], (::[(FiberId.Runtime, Any, Int)], Int)]
+  private[zio] val fiberRefLocals: Map[FiberRef[_], FiberRefs.Value]
 ) { self =>
+  import zio.FiberRefs.{StackEntry, Value}
 
   /**
    * Returns a new fiber refs with the specified ref deleted from it.
    */
   def delete(fiberRef: FiberRef[_]): FiberRefs =
-    FiberRefs(fiberRefLocals - fiberRef)
+    if (!fiberRefLocals.contains(fiberRef)) self
+    else FiberRefs(fiberRefLocals - fiberRef)
```

Comment:
In the worst-case scenario (ie. the `fiberRefLocals` contains the key), aren't we looking up twice for the key? 🤔

Reply:
Good point; I only just realised that `-` (i.e., `removed`) returns the same instance of `Map` if the map wasn't modified. So we can use `eq` after calling `-` to check if the map was modified or not